core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts

```diff
@@ -523,4 +554,173 @@ describe('Lesson Information card modal component', () => {
 
     expect(componentInstance.getProgressPercentage()).toEqual('28');
   });
+
+  it('should get checkpoints card indexes', () => {
+    const numCards = 5;
+    spyOn(playerTranscriptService, 'getNumCards').and.returnValue(numCards);
+
+    let cards: StateCard[] = [];
+    for (let i = 0; i < numCards; i++) {
+      cards.push({
+        getStateName: jasmine.createSpy('getStateName').and.returnValue(i),
+      });
+    }
+
+    spyOn(playerTranscriptService, 'getCard').and.callFake((index: number) => {
+      return cards[index];
+    });
+
+    spyOn(explorationEngineService, 'getStateFromStateName').and.callFake(
+      (stateName: string) => {
+        return {cardIsCheckpoint: parseInt(stateName) % 2 === 0};
+      }
+    );
+
+    const result = componentInstance.getCheckpointsCardIndexs();
+    expect(result).toEqual([0, 2, 4]);
+  });
+
+  it('should be able to return to all checkpoints', () => {
+    spyOn(playerPositionService, 'setDisplayedCardIndex').and.callThrough();
+    spyOn(playerPositionService.onActiveCardChanged, 'emit');
+    const closeSpy = spyOn(ngbActiveModal, 'close');
+
+    const checkpointsCardsIndexs = [1, 2, 5, 10];
+    const getCheckpointsCardIndexsSpy = spyOn(
+      componentInstance,
+      'getCheckpointsCardIndexs'
+    ).and.returnValue(checkpointsCardsIndexs);
+
+    for (let i = 0; i < checkpointsCardsIndexs.length; i++) {
+      const cardIndex = checkpointsCardsIndexs[i];
+      componentInstance.returnToCheckpoint(i);
+      expect(getCheckpointsCardIndexsSpy).toHaveBeenCalled();
+      expect(playerPositionService.setDisplayedCardIndex).toHaveBeenCalledWith(
+        cardIndex
+      );
+      expect(playerPositionService.onActiveCardChanged.emit).toHaveBeenCalled();
+      expect(playerPositionService.getDisplayedCardIndex()).toEqual(cardIndex);
+      expect(closeSpy).toHaveBeenCalled();
+    }
+  });
+
+  it('should return to the specified checkpoint', () => {
```

Comment:
I would remove this test, as it covers strictly less things than the test above (if I understand it correctly). It's better to have readable tests, and the more LOCs of tests you have, usually the less readable they are.

Reply:
Test removed.